torch/csrc/utils/tensor_new.cpp

```diff
@@ -612,8 +612,14 @@ Tensor sparse_csr_tensor_ctor(c10::DispatchKey dispatch_key, at::ScalarType scal
   const int NUM_ARGS = 9, CROW_INDICES_ARG = 0, COL_INDICES_ARG = 1, VALUES_ARG = 2;
   ParsedArgs<NUM_ARGS> parsed_args;
   auto r = parser.parse(args, kwargs, parsed_args);
-  THPObjectPtr crow_indices_dtype_attr(PyObject_GetAttrString(r.pyobject(CROW_INDICES_ARG), "dtype"));
-  THPObjectPtr col_indices_dtype_attr(PyObject_GetAttrString(r.pyobject(COL_INDICES_ARG), "dtype"));
+  auto safe_get_attr_string = [](PyObject *o, const char *attr_name) -> PyObject* {
+    // Check that attribute exists before getting its value
+    // Subsequent Python C API calls might return bogus values if exception is not handled
+    // See https://github.com/pytorch/pytorch/issues/58520 for more details
+    return PyObject_HasAttrString(o, attr_name) ? PyObject_GetAttrString(o, attr_name): nullptr;
```

Comment:
nit: You can actually do the get and clear the error if it returned null to be marginally faster. hasattr is already doing that internally: https://github.com/python/cpython/blob/1270ad6ec8fe1f71e7d88086474f96f99670ac3a/Objects/object.c#L820-L830

So the updated diff might just be:
```diff
THPObjectPtr crow_indices_dtype_attr(PyObject_GetAttrString(r.pyobject(CROW_INDICES_ARG), "dtype"));
+ if (!crow_indices_dtype_attr) PyErr_Clear();
THPObjectPtr col_indices_dtype_attr(PyObject_GetAttrString(r.pyobject(COL_INDICES_ARG), "dtype"));
+ if (!col_indices_dtype_attr) PyErr_Clear();
```

Reply:
Good point, updated lambda to clear error.